### PR TITLE
Add alternative parametrization for ShellSpatialModel

### PR DESCRIPTION
**Pull request by QRemy**
This PR add an alternative parametrization for the ShellSpatialModel using outer radius and relative width, so eta=width/Rout  act as a shape parameter similarly than for the GeneralizedGaussian.

### CODE REVIEW

File: gammapy/modeling/models/spatial.py [hunk 1]
```diff
@@ -159,11 +159,6 @@ def integrate_geom(self, geom):
 
         return Map.from_geom(geom=geom, data=data.value, unit=data.unit)
 
-    @property
-    def evaluation_radius(self):
```
Comment (adonath) [inline gammapy/modeling/models/spatial.py:163]: Why did you remove this? The purpose is to provide a default for custom implemented model, so that users are not required to implement an evaluation radius. But it was obviously missing a unit test...

File: gammapy/modeling/models/spatial.py [hunk 2]
```diff
@@ -518,7 +513,7 @@ def evaluation_radius(self):
 
     def to_region(self, **kwargs):
         """Model outline (`~regions.EllipseSkyRegion`)."""
-        minor_axis = Angle(self.r_0.quantity * (1 - self.e.quantity))
+        minor_axis = Angle(self.r_0.quantity * np.sqrt(1 - self.e.quantity ** 2))
```
Comment (adonath) [inline gammapy/modeling/models/spatial.py:516]: If the previous implementation was incorrect, can you please add a regression test, so that change is covered by a test?

File: gammapy/modeling/models/tests/test_spatial.py [hunk 3]
```diff
@@ -224,6 +225,16 @@ def test_sky_shell():
     assert_allclose(radius.value, rad.value + width.value)
     assert isinstance(model.to_region(), CircleAnnulusSkyRegion)
 
+    model = Shell2SpatialModel(lon_0="1 deg", lat_0="45 deg", r_0=rad + width, eta=0.5)
```
Comment (adonath) [inline gammapy/modeling/models/tests/test_spatial.py:228]: Please introduce a separate test for the model...

File: gammapy/utils/regions.py [hunk 4]
```diff
@@ -148,7 +148,14 @@ def f(x, coords):
         center = SkyCoord(lon * u.deg, lat * u.deg)
         return np.sum(center.separation(coords).deg)
 
-    result = minimize(f, x0=[0, 0], args=(positions,), bounds=[(-180, 180), (-90, 90)])
+    eps = 1e-5
```
Comment (adonath) [inline gammapy/utils/regions.py:151]: This change seems reasonable, but unrelated? Did you see any test fails that motivate the change?

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @QRemy, I have left some minor comments. I would propose to add a separate entry in the model Gallery for the `Shell2SpatialModel`, including the formula for the parametrisation.

File: gammapy/modeling/models/spatial.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/models/spatial.py:163]: It was duplicated

File: gammapy/utils/regions.py (hunk 4 above)
Comment (QRemy) [reply gammapy/utils/regions.py:151]: yes it was failing on my machine

File: gammapy/modeling/models/spatial.py (hunk 1 above)
Comment (adonath) [reply gammapy/modeling/models/spatial.py:163]: Thanks, indeed! I missed that...

File: gammapy/utils/regions.py (hunk 4 above)
Comment (adonath) [reply gammapy/utils/regions.py:151]: OK, interesting...this fail didn't appear in any of our CI builds. I guess one could choose the bounds even more reasonable, as the point must lie inside the polygon spanned by the others, but I can do this in a follow up PR. So just leave as is...

File: gammapy/modeling/models/spatial.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/models/spatial.py:516]: added

File: examples/models/spatial/plot_shell2.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            \begin{cases}
+                \sqrt{r_{out}^2 - \theta^2} - \sqrt{r_{in}^2 - \theta^2} &
+                             \text{for } \theta \lt r_{in} \\
+                \sqrt{r_{out}^2 - \theta^2} &
+                             \text{for } r_{in} \leq \theta \lt r_{out} \\
+                0 & \text{for } \theta > r_{out}
+            \end{cases}
+
+where :math:`\theta` is the sky separation, :math:`r_{\text{out}}` is the outer radius
+and  :math:`r_{\text{in}}` is the inner radius.
```
Comment (adonath) [inline examples/models/spatial/plot_shell2.py:22]: One last comment: could you at the definition of `R_in` with respect to `eta` here?

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @QRemy, I have left one last comment,  once this is addressed this PR is ready to merge...

**Review by adonath — APPROVED** [review]
Thanks @QRemy, no further comments from my side!
